objectivec/GPBWellKnownTypes.m

```diff
@@ -115,3 +115,52 @@ - (void)setTimeIntervalSince1970:(NSTimeInterval)timeIntervalSince1970 {
 }
 
 @end
+
+NSString *const GPBTypeGoogleApisComPrefix = @"type.googleapis.com/";
+
+@implementation GPBAny (GBPWellKnownTypes)
+
+- (NSString*)typeName {
+  NSAssert([self.typeURL hasPrefix:GPBTypeGoogleApisComPrefix],
+           @"Invalid any type url (%@).", self.typeURL);
+  if (![self.typeURL hasPrefix:GPBTypeGoogleApisComPrefix]) {
+    return @"";
```

Comment:
you rarely if ever need to check for nil in Objective C. A message to nil just returns nil/0.